src/dredd.coffee

```diff
@@ -28,37 +28,85 @@ class Dredd
         duration: 0
     @configuration = applyConfiguration(config, @stats)
     configureReporters @configuration, @stats, @tests
-    @runner = new Runner(@configuration)
+    @runner = new Runner @configuration
 
   run: (callback) ->
     config = @configuration
     stats = @stats
 
-    fs.readFile config.blueprintPath, 'utf8', (loadingError, data) ->
-      return callback(loadingError, stats) if loadingError
-      reporterCount = config.emitter.listeners('start').length
-      config.emitter.emit 'start', data, () ->
-        reporterCount--
-        if reporterCount is 0
-          protagonist.parse data, blueprintParsingComplete
-
-    blueprintParsingComplete = (protagonistError, result) =>
-      return callback(protagonistError, config.reporter) if protagonistError
-      
-      if result['warnings'].length > 0
-        for warning in result['warnings']
-          message = 'Parser warning: ' + ' (' + warning.code + ') ' + warning.message
-          for loc in warning['location']
-            pos = loc.index + ':' + loc.length
-            message = message + ' ' + pos
-          logger.warn message
-      
-      runtime = blueprintAstToRuntime result['ast']
-      runtimeError = handleRuntimeProblems runtime
-      return callback(runtimeError, stats) if runtimeError
-
-      @runner.run runtime['transactions'], () =>
-        @transactionsComplete(callback)
+    config.files = []
+
+    # expand all globs
```

Comment:
This is just my opinion, but I think this section would be more readable if every part that starts with a `# descriptive name` comment were replaced with a call to `descriptiveName()` function. A couple of cents from the peanut gallery.